actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -79,6 +79,37 @@ def form_tag(url_for_options = {}, options = {}, &block)
         end
       end
 
+      # Generate an HTML <tt>id</tt> attribute value for the given name and
+      # field combination
+      #
+      # Return the value generated by the <tt>FormBuilder</tt> for the given
+      # attribute name.
+      #
+      #   <%= label_tag :post, :title %>
+      #   <%= text_field_tag :post, :title, aria: { describedby: field_id(:post, :title, :error) } %>
+      #   <%= tag.span("is blank", id: field_id(:post, :title, :error) %>
+      #
+      # In the example above, the <tt><input type="text"></tt> element built by
+      # the call to <tt>text_field_tag</tt> declares an
+      # <tt>aria-describedby</tt> attribute referencing the <tt><span></tt>
+      # element, sharing a common <tt>id</tt> root (<tt>post_title</tt>, in this
+      # case).
+      def field_id(object_name, method_name, *suffixes, index: nil)
+        if object_name.respond_to?(:model_name)
+          object_name = object_name.model_name.singular
+        end
+
+        sanitized_object_name = object_name.to_s.gsub(/\]\[|[^-a-zA-Z0-9:.]/, "_").delete_suffix("_")
+
+        sanitized_method_name = method_name.to_s.delete_suffix("?")
+
+        if sanitized_object_name.empty?
+          sanitized_method_name.dup
+        else
+          [sanitized_object_name, index, sanitized_method_name, *suffixes].compact.join("_")
```

Comment:
Should we avoid the [] and compact here by using string interpolation like the previous code?

Reply:
@rafaelfranca how might we combine the style of the previous code with the `suffixes` Array?

Reply:
```ruby
if sanitized_object_name.empty?
  sanitized_method_name.dup
elsif suffixes.empty?
  if index
    "#{sanitized_object_name}_#{index}_#{sanitized_method_name}"
  else
     "#{sanitized_object_name}_#{sanitized_method_name}"
  end
else
  [sanitized_object_name, index, sanitized_method_name, *suffixes].compact.join("_")
end
```